### DISCUSSION TITLE
Does ArgoCD support Redis in Cluster mode?

### DISCUSSION THREAD

**Discussion by ramanNarasimhan77** [argoproj/argo-cd#20729] [Q&A]
My organization has an inhouse Redis chart, and it deploys Redis in Cluster Mode.

Looking at https://github.com/argoproj/argo-cd/blob/master/util/cache/cache.go#L132, it seems that only Standalone/Sentinel mode is supported.

How can I get Argo-CD to work with an external Redis which is deployed in Cluster Mode?

Also, would using Universal client https://redis.uptrace.dev/guide/universal.html in Argo-CD be better so that it is agnostic to how redis is deployed (standalone or using a proxy, Sentinel, or cluster)?

**Comment by ramanNarasimhan77** [comment]
I found an open PR for this https://github.com/argoproj/argo-cd/pull/14001

**Comment by andrii-korotkov-verkada** [comment]
What do you want to use that external Redis for? If for the same things as ArgoCD uses Redis, I'd just try HA mode.

**Reply by ramanNarasimhan77** [reply]
@andrii-korotkov-verkada 
Yes that would work but in all our internal clusters we already have Redis operator installed, and using CRD we are creating custom resources that define redis clusters. 

In our kubernetes cluster, we install multiple charts like ArgoCD, Gitea (for use with Argocd as git service), and few other charts. And each of these services has dedicated Redis clusters. 

For HA, It would be great if I can reuse the operator/crd setup itself instead of having to maintain another version of redis chart with Sentinel and optionally HAProxy.

**Comment by remi-gelinas** [comment]
Was this ever moved forward? Google is promoting their Redis Cluster Memorystore over their normal instances, and Argo is the only workload we have that still needs a normal HA instance deployed for it.

**Comment by rafaeltovargarrido** [comment]
maybe you can use Redis cluster, but not directly adding a layer like an envoy should help. The main issue can be the operations argocd doesn't to redis because envoy doesn't support all of them like ADD.
